x-pack/plugins/actions/server/usage/actions_telemetry.ts

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { ISavedObjectsRepository } from 'kibana/server';
+import { TaskManagerStartContract } from '../../../task_manager/server';
+import { ActionTypeRegistry } from '../action_type_registry';
+
+export async function getTotalCount(savedObjectsClient: ISavedObjectsRepository) {
+  const findResult = await savedObjectsClient.find({
+    type: 'action',
+  });
+
+  return findResult.total;
+}
+
+export async function getInUseTotalCount(savedObjectsClient: ISavedObjectsRepository) {
+  const findResult = await savedObjectsClient.find({
+    type: 'alert',
+    fields: ['actions'],
+  });
+
+  if (findResult.total === 0) {
+    return 0;
+  }
+
+  return new Set(
+    findResult.saved_objects.reduce(
+      (actionIds: string[], savedObj: any) =>
+        actionIds.concat(
+          savedObj.references
+            .filter((ref: any) => ref.type === 'action')
+            .map((action: any) => action.id)
+        ),
+      []
+    )
+  ).size;
+}
+
+export async function getTotalCountByActionTypes(
+  savedObjectsClient: ISavedObjectsRepository,
+  actionTypeRegistry: ActionTypeRegistry
+) {
+  const totalByActionType = actionTypeRegistry
+    .list()
+    .reduce(async (accPromise: any, actionType) => {
+      const acc = await accPromise;
+      const total = await getTotalCountByActionType(savedObjectsClient, actionType.id);
+      return { ...acc, [actionType.name]: total };
+    }, Promise.resolve({}));
+
+  return totalByActionType;
+}
+
+export async function getExecutions(taskManager: TaskManagerStartContract) {
```

Comment:
I think we'll end up switching this to use the eventLog to get the numbers.  We'll have to add an API to return counts without numbers (optimized for this usage).  I'm not exactly sure what the query. below returns, but that's fine, if we're replacing it with eventLog.

If so, suggest we add a TODO comment here, with a link to the relevant PR - https://github.com/elastic/kibana/issues/55633#issuecomment-592054816

Reply:
Yes, we will replace it with eventLog. I will add the proper TODO